### PR TITLE
added an installing section to README.md

### PR DESCRIPTION
**Pull request by cloutier**
I took some notes while installing my server. I polished them up, shamelessly stole a few sentences from you in the issues and made a PR out of it. I'll continue improving this as my understanding of the codebase increases.

I am not so sure about the hmac section, if my lenght of 128 is big enough and if I should recommend people to take a backup. What do you think?

I also apologize in advance for typos/bad grammar, english is a second language for me.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,48 @@ mode.
 * leveldb
 
 
+## Installing
+
+You will need a bitcoin node running on the same machine. You can safely use 
+the testnet. You will need this in your bitcoin.conf:
+```
+#testnet=1 # uncomment this for testnet
+txindex=1
```
Comment (petertodd) [inline README.md]: Actually this isn't needed!

In fact, you can even use a pruned node, which is what my alice and bob calendars do.

File: README.md [hunk 2]
```diff
@@ -14,6 +14,48 @@ mode.
...
+```
+
+You need to define the public URI of your calendar server in ~/.otsd/calendar/uri .
+Basically whatever is in that file is put into the URI field of pending attestations returned by that calendar server. So when a client goes to verify a timestamp created by that calendar, if the URI starts with http or https, it'll try to make a HTTP(S) connection to that server to fetch the rest of the timestamp proof.
+
+E.g. the alice.btc.calendar.opentimestamps.org calendar server has https://alice.btc.calendar.opentimestamps.org in the ~/.otsd/calendar/uri file.
+
+You will also need a hmac key:
+``` 
+openssl rand -base64 128 > .otsd/calendar/hmac-key
```
Comment (petertodd) [inline README.md]: Probably better to use /dev/urandom, and also, the hmac-key is used as raw binary, not hex. So change that to

    dd if=/dev/random of=~/.otsd/calendar/hmac-key bs=32 count=1

**Comment by escapedcat** [comment]
Maybe also mention that `~/.otsd/` will be created automatically after you run the server once. If that's correct. Might be clear for others but easier for people who don't know/expect this.

**Comment by cloutier** [comment]
Thanks for the feedback @petertodd and @escapedcat. I've updated the readme accordingly.

**Comment by sergeanthorst** [comment]
I think just commenting out the #testnet=1 line in the .bitcoin/bitcoin-conf is not enough to have the opentimestamp server run with the testnet. 
I did a little debugging and even with testnet=1 otsd tried to connect to the main net port. 
A workarround would be to add rpchost and rpcport to bitcoin-conf or to start otsd with the --btc-testnet parameter. 
Maybe this is also worth mentioning in the installation readme.

**Comment by petertodd** [comment]
@sergeanthorst For testnet, you need to tell otsd to use testnet with the `--btc-testnet` flag.

**Comment by petertodd** [comment]
Merged!

I used your new writeup to improve the README a little further with some further edits; thanks a lot!

**Pull request merged**